src/main/java/com/celements/webdav/SardineAdapter.java

```
@@ -0,0 +1,263 @@
+    try (InputStream is = getSardine(remoteLogin).get(url.toExternalForm())) {
+      byte[] content = IOUtils.toByteArray(is);
+      LOGGER.info("load - {} : {} bytes", url, content.length);
+      return content;
+    } catch (IOException exc) {
+      throw new WebDavException(url, remoteLogin, exc);
+    }
+  }
+
+  @Override
+  public boolean store(Path filePath, byte[] content) throws WebDavException {
+    return store(filePath, content, getConfiguredWebDavRemoteLogin());
+  }
+
+  @Override
+  public boolean store(Path filePath, byte[] content, RemoteLogin remoteLogin)
+      throws WebDavException {
+    boolean success = false;
+    URL url = buildCompleteUrl(remoteLogin, filePath);
+    try {
+      Sardine sardine = getSardine(remoteLogin);
+      if (!sardine.exists(url.toExternalForm())) {
+        getSardine(remoteLogin).put(url.toExternalForm(), content);
+        LOGGER.info("store - {}", url);
+        success = true;
+      } else {
+        LOGGER.debug("store - tried to overwrite [{}]", url);
+      }
+    } catch (SardineException exc) {
+      LOGGER.warn("store - failed on [{}] with login [{}]", url, remoteLogin, exc);
+    } catch (IOException exc) {
+      throw new WebDavException(url, remoteLogin, exc);
+    }
+    return success;
+  }
+
+  @Override
+  public boolean delete(Path path) throws WebDavException {
+    return delete(path, getConfiguredWebDavRemoteLogin());
+  }
+
+  @Override
+  public boolean delete(Path path, RemoteLogin remoteLogin) throws WebDavException {
+    boolean success = false;
+    URL url = buildCompleteUrl(remoteLogin, path);
+    try {
+      Sardine sardine = getSardine(remoteLogin);
+      if (sardine.exists(url.toExternalForm())) {
+        sardine.delete(url.toExternalForm());
+        LOGGER.info("delete - {}", url);
+        success = true;
+      } else {
+        LOGGER.debug("delete - tried to delete inexistent path [{}]", url);
+      }
+    } catch (SardineException exc) {
```

I think it is better to throw an exception in case the file does not exists. Similar to the Files.delete semantic of Java: https://docs.oracle.com/javase/tutorial/essential/io/delete.html
it is then easier for the client to decide if this is excepted and can be ignored or if this is a more dangerous situation which needs to be put to the user or logged as error (e.g. like a mutext-lock file got "lost").

---

👍